internal/metrics/metrics_thresholds.go

```diff
@@ -1,13 +1,12 @@
 package metrics
 
 import (
+	"cmp"
 	"time"
-
-	"golang.org/x/exp/constraints"
 )
 
 // Thresholds encapsulates a set of bucket thresholds used in Summary[T].
-type Thresholds[T constraints.Float | constraints.Integer] struct {
+type Thresholds[T cmp.Ordered] struct {
```

Comment:
Note: this is not equivalent, in also includes `string`.
It makes sense to revert it and add a `//nolint` annotation instead.
Ditto below.